client/opentracing/src/test/java/org/hawkular/apm/client/opentracing/SamplingTest.java

```diff
@@ -0,0 +1,185 @@
+        Tracer tracer = new APMTracer(traceRecorder, Sampler.NEVER_SAMPLE);
+
+        Span span = tracer.buildSpan("foo")
+                .start();
+
+        span.finish();
+        Assert.assertEquals(0, traceRecorder.getTraces().size());
+    }
+
+    @Test
+    public void testNeverSampleExtractedSampleAllContext() {
+        APMTracerTest.TestTraceRecorder traceRecorder = new APMTracerTest.TestTraceRecorder();
+        Tracer tracer = new APMTracer(traceRecorder, Sampler.NEVER_SAMPLE);
+
+        Span span = tracer.buildSpan("foo")
+                .asChildOf(extractedTraceState(tracer, ReportingLevel.All))
+                .start();
+
+        span.finish();
+        Assert.assertEquals(1, traceRecorder.getTraces().size());
+    }
+
+    @Test
+    public void testAlwaysSample() {
+        APMTracerTest.TestTraceRecorder traceRecorder = new APMTracerTest.TestTraceRecorder();
+        Tracer tracer = new APMTracer(traceRecorder, Sampler.ALWAYS_SAMPLE);
+
+        Span span = tracer.buildSpan("foo")
+                .start();
+
+        span.finish();
+        Assert.assertEquals(1, traceRecorder.getTraces().size());
+    }
+
+    @Test
+    public void testAlwaysSampleExtractedSampleNoneContext() {
+        APMTracerTest.TestTraceRecorder traceRecorder = new APMTracerTest.TestTraceRecorder();
+        Tracer tracer = new APMTracer(traceRecorder, Sampler.ALWAYS_SAMPLE);
+
+        Span span = tracer.buildSpan("foo")
+                .asChildOf(extractedTraceState(tracer, ReportingLevel.None))
+                .start();
+
+        span.finish();
+        Assert.assertEquals(0, traceRecorder.getTraces().size());
+    }
+
+    @Test
+    public void testSamplingPriorityChangedToZero() {
+        APMTracerTest.TestTraceRecorder traceRecorder = new APMTracerTest.TestTraceRecorder();
+        Tracer tracer = new APMTracer(traceRecorder, Sampler.ALWAYS_SAMPLE);
+
+        Span rootSpan = tracer.buildSpan("foo")
+                .asChildOf(extractedTraceState(tracer, ReportingLevel.All))
+                .start();
+
+        Span descendant = tracer.buildSpan("bar")
+                .asChildOf(rootSpan)
+                .start();
+
+        Map<String, String> carrier = new HashMap<>();
+        tracer.inject(descendant.context(), Format.Builtin.TEXT_MAP, new TextMapInjectAdapter(carrier));
+        Assert.assertEquals(ReportingLevel.All.name(), carrier.get(Constants.HAWKULAR_APM_LEVEL));
+
+        Span descendantZeroSampling = tracer.buildSpan("bar")
+                .asChildOf(rootSpan)
+                .start();
+
+        descendantZeroSampling.setTag(Tags.SAMPLING_PRIORITY.getKey(), 0);
+        carrier.clear();
+        tracer.inject(descendantZeroSampling.context(), Format.Builtin.TEXT_MAP, new TextMapInjectAdapter(carrier));
+        Assert.assertEquals(ReportingLevel.None.name(), carrier.get(Constants.HAWKULAR_APM_LEVEL));
+
+        descendantZeroSampling.finish();
+        descendant.finish();
+        rootSpan.finish();
+
```

Comment:
Could you add a check here that the trace fragment has been reported. And possibly clear the test tracer before starting the next span - and then make sure that next span doesn't result in a fragment being reported?